core/common/src/files/FileSystem.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.files
+
+import kotlinx.io.IOException
+import kotlinx.io.Sink
+import kotlinx.io.Source
+
+/**
+ * An interface providing basic operations on a file system.
+ *
+ * **This API is unstable and subject to change.**
+ */
+public interface FileSystem {
```

Comment:
Here and  below: let's make modality `sealed` to prevent implementation of interfaces/classes that we probably may want to evolve?

Reply:
Great idea, thanks!